src/applications/disability-benefits/new-686/config/chapters/report-add-a-spouse/current-spouse-marriage-history-details/currentSpouseMarriageHistoryDetails.js

```diff
@@ -0,0 +1,93 @@
+import currentOrPastDateUI from 'platform/forms-system/src/js/definitions/currentOrPastDate';
+
+import { genericSchemas } from '../../../generic-schema';
+import { spouseTitle } from './helpers';
+
+const { date, genericLocation, genericTextInput } = genericSchemas;
+
+const reasonMarriageEndedSchema = {
+  type: 'string',
+  enum: ['DIVORCE', 'DEATH', 'ANNULMENT', 'OTHER'],
+  enumNames: ['Divorce', 'Death', 'Annulment', 'Other'],
+};
+
+const reasonMarriageEndedUISchema = {
+  'ui:required': formData => formData.spouseWasMarriedBefore === true,
+  'ui:title': 'Why did marriage end?',
+  'ui:widget': 'radio',
+};
+
+export const schema = {
+  type: 'object',
+  properties: {
+    spouseMarriageHistory: {
+      type: 'array',
+      items: {
+        type: 'object',
+        properties: {
+          marriageStartDate: date,
+          marriageStartLocation: genericLocation,
+          reasonMarriageEnded: reasonMarriageEndedSchema,
+          reasonMarriageEndedOther: genericTextInput,
+          marriageEndDate: date,
+          marriageEndLocation: genericLocation,
+        },
+      },
+    },
+  },
+};
+
+export const uiSchema = {
+  spouseMarriageHistory: {
+    items: {
+      'ui:title': spouseTitle,
+      marriageStartDate: {
+        ...currentOrPastDateUI('Date of marriage'),
+        ...{
+          'ui:required': formData => formData.spouseWasMarriedBefore === true,
```

Comment:
NIT: If `spouseWasMarriedBefore` is always boolean, is it necessary to compare it to `true`? 

Would either of the following work?
```
// if `spouseWasMarriedBefore` is guaranteed to be a boolean, then return the boolean
- 'ui:required': formData => formData.spouseWasMarriedBefore === true,
+ 'ui:required': formData => formData.spouseWasMarriedBefore,
```
or
```
// if `spouseWasMarriedBefore` could be `null` or `undefined`, then coerce the truthiness of it
- 'ui:required': formData => formData.spouseWasMarriedBefore === true,
+ 'ui:required': formData => !!formData.spouseWasMarriedBefore,
```

Reply:
Refactored [here](https://github.com/department-of-veterans-affairs/vets-website/pull/11820/commits/ff8536a2d81259a55660938c1de9296259bda912)